### ISSUE TITLE
Create a class for RichConfigurableFields

### ISSUE THREAD

**Issue by seanstory** [elastic/connectors#1673]
### Problem Description
It's [been noted](https://github.com/elastic/connectors-python/pull/1653#discussion_r1334046142) that RCF's rely on "magic strings" instead of enums or properties that clearly define which fields should be set for new configurations and should be evaluated in logic that handles the configuration objects. In fact, we rarely handle them as "objects" but instead as python dicts. This could be more object-oriented.

### Proposed Solution
- [ ] introduce a class for RichConfigurableField
- [ ] have it define constants or enums for the fields it requires
- [ ] replace "magic string" usages of strings like `"value"` and `"default_value"` etc with constant/enum usage

**Comment by nick2432** [comment]
can i work on this?

**Comment by seanstory** [comment]
Absolutely! Make sure you've read through https://github.com/elastic/connectors/blob/main/docs/CONTRIBUTING.md so you know what you're getting into. But our team welcomes external contributions! 

If you do put in the work, make sure you've registered with the Elastic Contributor Program to get rewards! https://www.elastic.co/community/contributor

**Comment by srinchow** [comment]
I don't see this assigned to anyone if there is no work done behind the scenes for the same can i pick this @seanstory

**Comment by seanstory** [comment]
Go for it! Same guidance as above. :)

**Comment by artem-shelkovnikov** [comment]
Voting to close this one for now - doesn't seem there's a need for it now
